File: go/raindrops/raindrops.go
package raindrops

import "strconv"

// Convert returns a message based on its factors
func Convert(number int) string {
	stringToReturn := ""
	if number%3 == 0 {
		stringToReturn += "Pling"
	}
	if number%5 == 0 {
		stringToReturn += "Plang"
	}
	if number%7 == 0 {
		stringToReturn += "Plong"
	}
	if stringToReturn == "" {
		return strconv.Itoa(number)
	}
	return stringToReturn
}

File: fsharp/raindrops/Raindrops.fs
module Raindrops

let convert (number: int): string = 
     match number%3, number%5, number%7 with      
      | 0,0,0 -> "PlingPlangPlong"
      | 0,0,_ -> "PlingPlang"
      | 0,_,0 -> "PlingPlong"
      | _,0,0 -> "PlangPlong"
      | 0,_,_ -> "Pling"
      | _,0,_ -> "Plang"
      | _,_,0 -> "Plong"
      | _,_,_ -> number.ToString()